### ISSUE TITLE
mapa

### ISSUE THREAD

**Issue by jacekkarczmarczyk** [szczepienia/szczepienia.github.io#17]
mapa ze wszystkimi wyfiltrowanymi punktami byłaby dużo wygodniejsza jeśli chodzi o znalezienie bliskiego punktu

**Comment by szczepienia** [comment]
Tak, myśle o tym jak zobaczyłem na vaccinespotter.org :)
Tylko trochę czasu potrzebuję

**Comment by szczepienia** [comment]
Pamiętam o tym, ale to chyba największy ficzer do zrobienia (i najbardziej mnie interesujący też :), ale mam jeszcze pracę i rodzinę, więc zostaje mało czasu na hobby :)

**Comment by szczepienia** [comment]
Jeszcze pytanie do osób zorientowanych, jaka mapa ma duży darmowy tier? Mapbbox wygląda ok ale za darmo jest do 50 tys. wyświtleń w miesiącu, co zniknie chyba w jeden dzień :(
Mapy googlowe są chyba jeszcze bardziej restrykcyjne.

**Comment by Daste745** [comment]
OpenStreetMap w połączeniu z OpenLayers są teoretycznie nieograniczone. Na [wiki OSM](https://wiki.openstreetmap.org/wiki/Openlayers_POI_layer_example) mają przykład jak zrobić taką mapę a w [dokumentacji OpenLayers](http://dev.openlayers.org/docs/files/OpenLayers/Layer/Text-js.html) też jest masa przydatnych rzeczy. Jeśli będzie potrzeba to mogę pomóc z implementacją takiej mapki.

**Comment by szczepienia** [comment]
@Daste745 Dzięki, wygląda całkiem prosto (przynajmniej wersja bez filtrowania), tylko problem może być z tile serverem, bo ten z OSM nie chce "heave use", nie wiem co się pod to łapie, narazie zadałem pytanie mailem.

**Comment by themysteq** [comment]
@szczepienia może cache'owanie Tile tutaj zrobi robotę - wszak mapa często się zmieniać nie będzie jeśli chodzi o podkład, więc można to agresywnie cache'ować.
https://github.com/yagajs/leaflet-cached-tile-layer ale to tak na szybko.
cc: @Daste745 

Sam mogę pomóc jedynie przy devopsowaniu lub wrzuceniu kilku groszy na donate ;)

**Comment by lukaszbel** [comment]
Osm w teorii nie zezwala na hostowanie ciezkich serwisow bezposrednio od siebie. Moga zbanowac, jak przegniecie. Alternatywa jest pobranie wektorowych danych z OSM i postawienie wlasnego tile servera na np. mapniku, ale trzeba wydajna maszyne. Zerknijcie tez na rozwiazania firmy Klokan https://openmaptiles.org/ platne, ale zapewnia mapy wektorowe, nie tak ciezkie jak rastry.
Open Layers jest ok, choc lzejszy nest Leaflet. Jakby co, siedze w tym na co dzien (jako analityk GIS nie dev) i moge merytorycznie doradzic.

**Comment by themysteq** [comment]
Tileserver można postawić samodzielnie lub zrobić publiczny cache tiles'ów OSMowych:

Pierwszy wynik z brzegu.
https://coderwall.com/p/--wgba/nginx-reverse-proxy-cache-for-openstreetmap
Ja nie dam rady tego zrobić dzisiaj niestety

**Comment by dblachut** [comment]
@szczepienia obecnie jest możliwość podania lokalizacji, fajnie by było jakby w tabeli była wtedy dodatkowa kolumna z odległością do danego punktu.

**Comment by szczepienia** [comment]
Tak, miałem już takie zgłoszenie, tylko mam problem gdzie dodać taką odległość.

Na komórkach (większość użytkowników) jest miejce na 3 kolumny obecnie (miasto, data i godziny). I nie wiem gdzie wcisnąć odległość.

Obecnie mam pomysł aby dorzucić to wtedy do Miasta i zrobić sortowanie po odległości od razu gdy ktoś zaktualizuje swoją pozycję.

**Comment by dblachut** [comment]
Chyba w poziomym layoucie powinno się zmieścić, a poza tym scrollowanie w bok chyba nie jest tak uciążliwe. 😉

**Comment by szczepienia** [comment]
Wejdzie jak tylko github pages zacznie działać.

**Comment by szczepienia** [comment]
i działa, jest sortowanie po odległości :)
